lib/ansible/modules/windows/win_firewall_rule.ps1

```diff
@@ -54,351 +72,390 @@ function preprocessAndCompare($key, $outputValue, $fwsettingValue) {
+            $result.output = Invoke-Expression $command | Where { $_ }
+            $rc = $LASTEXITCODE
+        }
+        if ($rc -eq 0) {
+            if ($diff_support) {
+                $result.diff.after = ConvertTo-SortedKV $fwsettings
+                $result.diff.before= ""
+            }
+            $result.changed = $true
+            $result.msg += @("Created firewall rule '$name'")
+        } else {
+            $result.failed = $true
+            $result.msg += @("Create command '$command' failed with rc=$rc")
+        }
     } catch [Exception]{
-        $msg=@("Failed to create the rule")
-        $result=@{
-            output=$output
-            failed=$true
-            error=$_.Exception.Message
-            msg=$msg
-        };
-    };
+        $result.error = $_.Exception.Message
+        $result.failed = $true
+        $result.msg = @("Failed to create the rule '$name'")
+    }
     return $result
-};
+}
 
 function removeFireWallRule ($fwsettings) {
-    $msg=@()
+    $result = @{
+        changed = $false
+        diff = @{}
+        failed = $false
+        msg = @()
+    }
+
+    $command = "netsh advfirewall firewall delete rule name='$($fwsettings.'Rule Name')'"
     try {
-        $rawoutput=@(netsh advfirewall firewall delete rule name="$($fwsettings.'Rule Name')")
-        $rawoutput | Where {$_ -match '^([^:]+):\s*(\S.*)$'} | Foreach -Begin {
-                $FirstRun = $true;
-                $HashProps = @{};
-            } -Process {
-                if (($Matches[1] -eq 'Rule Name') -and (!($FirstRun))) {
-                    $output=$HashProps;
-                    $HashProps = @{};
-                };
-                $HashProps.$($Matches[1]) = $Matches[2];
-                $FirstRun = $false;
-            } -End {
-                $output=$HashProps;
-            };
-        $msg+=@("Removed the rule")
-        $result=@{
-            failed=$false
-            changed=$true
-            msg=$msg
-            output=$output
-        };
-    } catch [Exception]{
-        $msg+=@("Failed to remove the rule")
-        $result=@{
-            failed=$true
-            error=$_.Exception.Message
-            msg=$msg
+        $rc = 0
+        if (-not $check_mode) {
+            $result.output = Invoke-Expression $command | Where { $_ }
+            $rc = $LASTEXITCODE
+            $result.output | Where {$_ -match '^([^:]+):\s*(\S.*)$'} | Foreach -Begin {
+                    $FirstRun = $true
+                    $HashProps = @{}
+                } -Process {
+                    if (($Matches[1] -eq 'Rule Name') -and (-not $FirstRun)) {
+                        $result.output = $HashProps
+                        $HashProps = @{}
+                    }
+                    $HashProps.$($Matches[1]) = $Matches[2]
+                    $FirstRun = $false
+                } -End {
+                    $result.output = $HashProps
+                }
+        }
+        if ($rc -eq 0 -or $rc -eq 1) {
```

Comment:
To be honest, I don't know or remember :-/
However, the integration tests do remove the same rule twice to cover this situation, and it is doing the right thing.